apps/ios/Shared/Views/Chat/ChatItemView.swift

```diff
@@ -131,52 +144,60 @@ struct ChatItemContentView<Content: View>: View {
     }
 
     private func chatFeatureView(_ feature: Feature, _ iconColor: Color) -> some View {
-        CIChatFeatureView(chatItem: chatItem, feature: feature, iconColor: iconColor)
+        CIChatFeatureView(chatItem: chatItem, revealed: $revealed, feature: feature, iconColor: iconColor)
     }
 
-    private var membersConnectedItemText: Text {
-        if let t = membersConnectedText {
-            return chatEventText(t, chatItem.timestampText)
+    private var mergedGroupEventText: Text? {
+        let (count, ns) = chatModel.getConnectedMemberNames(chatItem)
+        let members: LocalizedStringKey =
+            switch ns.count {
+            case 1: "\(ns[0]) connected"
+            case 2: "\(ns[0]) and \(ns[1]) connected"
+            case 3: "\(ns[0] + ", " + ns[1]) and \(ns[2]) connected"
+            default:
+                ns.count > 3
+                ? "\(ns[0]), \(ns[1]) and \(ns.count - 2) other members connected"
+                : ""
+            }
+        return if count <= 1 {
+            nil
+        } else if ns.count == 0 {
+            Text("\(count) group events")
+        } else if count > ns.count {
+            Text(members) + Text(" ") + Text("and \(count - ns.count) other events")
```

Comment:
might make it easier to not mess up grammar in translations..
```suggestion
            Text(members) + Text(", ") + Text("\(count - ns.count) other events")
```